content/docs/lists-and-keys.md

```diff
@@ -119,34 +121,34 @@ const todoItems = todos.map((todo) =>
 );
 ```
 
-When you don't have stable IDs for rendered items, you may use the item index as a key as a last resort:
+Quand vous n'avez pas d'ID stable pour les éléments rendus, vous pouvez utiliser l'index de l'élément en dernier recours :
 
 ```js{2,3}
 const todoItems = todos.map((todo, index) =>
-  // Only do this if items have no stable IDs
+  // Faites uniquement ceci si les éléments n'ont pas d'ID stable
   <li key={index}>
     {todo.text}
   </li>
 );
 ```
 
-We don't recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state. Check out Robin Pokorny's article for an [in-depth explanation on the negative impacts of using an index as a key](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). If you choose not to assign an explicit key to list items then React will default to using indexes as keys.
+Nous ne recommandons pas d'utiliser l'index comme clé si l'ordre des éléments change. Cela peut avoir un effet négatif sur les performances, et peut causer des problèmes avec l'état du composant. Vous pouvez lire l'article de Robin Pokorny pour une [explication en profondeur de l'impact négatif de l'utilisation de l'index comme clé](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318) (en anglais). Si vous choisissez de ne pas donner explicitement de clé aux éléments d'une liste, React utilisera les index par défaut.
```

Comment:
```suggestion
Nous vous recommandons de ne pas utiliser l'index comme clé si l'ordre des éléments est susceptible de changer. Ça peut avoir un effet négatif sur les performances, et causer des problèmes avec l'état du composant. Vous pouvez lire l'article de Robin Pokorny pour une [explication en profondeur de l'impact négatif de l'utilisation de l'index comme clé](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318) (en anglais). Si vous choisissez de ne pas donner explicitement de clé aux éléments d'une liste, React utilisera l’index par défaut.
```

- L’anglais *don’t recommend* est un euphémisme courant pour *recommend not to* : ce n’est pas tant qu’on ne *recommande pas* : on *recommande de ne pas*, ce qui est différent.
- Quelques retraits de répétitions superflues
- Bien d'avoir pensé à préciser la langue cible
- Tu as eu raison de garder *index* au singulier avant, on va rester là-dessus pour la dernière phrase.